kong/runloop/events.lua

```diff
@@ -261,17 +261,19 @@ end
 local function crud_snis_handler(data)
   log(DEBUG, "[events] SNI updated, invalidating cached certificates")
 
-  local sni = data.old_entity or data.entity
-  local sni_name = sni.name
-  local sni_wild_pref, sni_wild_suf = certificate.produce_wild_snis(sni_name)
-  core_cache:invalidate("snis:" .. sni_name)
-
-  if sni_wild_pref then
-    core_cache:invalidate("snis:" .. sni_wild_pref)
-  end
+  -- data.entity won't be nil so we put it in the front
+  for _, sni in ipairs { data.entity, data.old_entity } do
+    local sni_name = sni.name
+    local sni_wild_pref, sni_wild_suf = certificate.produce_wild_snis(sni_name)
+    core_cache:invalidate("snis:" .. sni_name)
+
+    if sni_wild_pref then
+      core_cache:invalidate("snis:" .. sni_wild_pref)
+    end
 
-  if sni_wild_suf then
-    core_cache:invalidate("snis:" .. sni_wild_suf)
+    if sni_wild_suf then
+      core_cache:invalidate("snis:" .. sni_wild_suf)
+    end
```

Comment:
We should skip invalidating if the strings are the same.

Reply:
done